src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -497,7 +529,14 @@ private void configureNetworkTables() {
 
 	@Override
 	public void periodic() {
-		pose = poseEstimator.update(gyroscope.getAngle(), getModulePositions());
+		Pose2d combinedPose, odometryPose;
+		synchronized (poseEstimator) {
+			combinedPose = poseEstimator.update(gyroscope.getAngle(), getModulePositions());
+		}
+		odometryPose = drivebaseOnlyOdometry.update(gyroscope.getAngle(), getModulePositions());
```

Comment:
Consider reading the gyroscope & module positions one, then assigning them twice